frontend_tests/node_tests/alert_words_ui.js

```diff
@@ -16,27 +16,27 @@ alert_words.initialize({
     alert_words: ["foo", "bar"],
 });
 
-run_test("render_alert_words_ui", ({mock_template}) => {
-    const word_list = $("#alert_words_list");
-    const appended = [];
-    word_list.append = (rendered) => {
-        appended.push(rendered);
-    };
-
-    const alert_word_items = $.create("alert_word_items");
-    word_list.set_find_results(".alert-word-item", alert_word_items);
-
-    alert_word_items.remove = () => {};
-
-    mock_template("settings/alert_word_settings_item.hbs", false, (args) => "stub-" + args.word);
+run_test("render_alert_words_ui", ({override_rewire}) => {
+    let render_alert_words_ui_called = false;
+    override_rewire(alert_words_ui, "render_alert_words_ui", () => {
+        const opts = alert_words.get_word_list();
+        assert.deepEqual(opts, [
+            {
+                word: "foo",
+            },
+            {
+                word: "bar",
+            },
+        ]);
+        render_alert_words_ui_called = true;
+    });
+
+    alert_words_ui.reset();
+    assert.equal(alert_words_ui.loaded, false);
 
-    const new_alert_word = $("#create_alert_word_name");
-    assert.ok(!new_alert_word.is_focused());
-
-    alert_words_ui.render_alert_words_ui();
-
-    assert.deepEqual(appended, ["stub-bar", "stub-foo"]);
-    assert.ok(new_alert_word.is_focused());
+    alert_words_ui.set_up_alert_words();
+    assert.equal(alert_words_ui.loaded, true);
+    assert.ok(render_alert_words_ui_called);
```

Comment:
This test basically tests `get_word_list()` in the override part and checks whether `render_alert_words_ui` is called which will always be called when `set_up_alert_words` is called. I don't think this test is much useful.

So, I think instead we should just write a simple test here that `render_alert_words_ui` returns early when loaded is false and mock `list_widget.create` function properly to make sure other tests pass since `list_widget.js` already has 100% coverage and has proper node tests.

Reply:
@sahil839 , I've made all the required changes you've asked for and added the node-test case which runs successfully now.

Thanks for help !